misc-utils/lsfd-sock-xinfo.c

```diff
@@ -52,41 +55,127 @@ static void load_xinfo_from_proc_udp6(ino_t netns_inode);
 static void load_xinfo_from_proc_udplite6(ino_t netns_inode);
 static void load_xinfo_from_proc_raw6(ino_t netns_inode);
 static void load_xinfo_from_proc_netlink(ino_t netns_inode);
+static void load_xinfo_from_proc_packet(ino_t netns_inode);
 
 static int self_netns_fd = -1;
 static struct stat self_netns_sb;
 
 static void *xinfo_tree;	/* for tsearch/tfind */
 static void *netns_tree;
 
+struct iface {
+	unsigned int index;
+	char name[IF_NAMESIZE];
```

Comment:
`IF_NAMESIZE + 1` for the trailing `0`?

Reply:
It seems that IF_NAMESIZE in kernel includes `\0`.

```
# ip link add name 0123456789abcdef type veth 
Error: argument "0123456789abcdef" is wrong: "name" not a valid ifname
# ip link add name 0123456789abcde type veth 
(no error)
$ expr length 0123456789abcde
15
```
linux/include/uapi/linux/if.h:
```
#define	IFNAMSIZ	16
```

To avoid risks, I did:
```
		strncpy(iface->name, ifa->ifa_name, sizeof(iface->name) - 1);
		iface->name[sizeof(iface->name) - 1] = '\0';
```

Reply:
> It seems that IF_NAMESIZE in kernel includes \0.

Thanks for checking that. I only looked at some derived defines in /usr/include and they did `IF_NAMESIZE + 1`